### PR TITLE
Some new devices and recipes

### PR DESCRIPTION
**Pull request by byorgey**
- Add quartz and quartz mines
- Add `clock`, `silicon`, and `circuit`
- Add `comparator` device
- Metal drills can now drill through boulders too
- Add pixels to the world, and recipes for `camera` and `scanner`

See discussion at https://github.com/swarm-game/swarm/issues/26#issuecomment-1152743741 and https://github.com/swarm-game/swarm/pull/361 .

### CODE REVIEW

**Comment by xsebek** [comment]
> Metal drills can now drill through boulders too.

Hah, thanks :+1:

File: data/recipes.yaml [hunk 1]
```diff
@@ -44,6 +44,18 @@
   out:
   - [1, wooden gear]
 
+- in:
+  - [1, board]
+  - [2, wooden gear]
+  out:
+  - [1, seesaw]
```
Comment (xsebek) [inline data/recipes.yaml:51]: Wouldn't it be better to use iron for a seesaw? It feels strange that we won't use it for cutting boards and stuff. :confused: 

I would not mind having a scale just from rocks and a branch, but some joke entity would be fun. :thinking:

Comment (byorgey) [reply data/recipes.yaml:51]: A seesaw is not for cutting: https://en.wikipedia.org/wiki/Seesaw  It is supposed to be a joke... :smile:   Maybe `seesaw` is a US-specific term?  We could call it a `teeter-totter` instead, maybe that sounds funnier anyway.

Comment (byorgey) [reply data/recipes.yaml:51]: Also, in terms of gameplay, I think a `comparator` should not be too hard to make (in particular I don't think it should require any iron).  They come in very handy.  You need a comparator if you want to do basic things like recurse on an `int` (so you can tell when it reaches 0).  I also do things like set up a robot to accept harvested trees, and when it notices that the number of trees is over a certain amount, then it crafts them into various products and sends them along.

Comment (TristanCacqueray) [reply data/recipes.yaml:51]: `teeter-totter` sounds better, I also thought seesaw was something for cutting stuff :)

Comment (byorgey) [reply data/recipes.yaml]: OK, done!

**Comment by byorgey** [comment]
The `clock` will probably end up being an ingredient for other things (e.g. a `time machine`, see #133), but I'd like to make it a device in its own right.  We could make it provide the ability to `wait`.  We could also create an instruction `time` that gets the absolute number of ticks since the beginning of the game, perhaps.  But I think I might do that in a separate PR.

**Pull request merged**